### ISSUE TITLE
Failing to sample from Dirichlet Process Mixture Model using SGLD

### ISSUE THREAD

**Issue by Habush** [blackjax-devs/blackjax#259]
### Bug Description

I'm trying to reproduce the example from tensorflow-probability titled [Fitting Dirichlet Process Mixture Model Using Preconditioned Stochastic Gradient Langevin Dynamics](https://www.tensorflow.org/probability/examples/Fitting_DPMM_Using_pSGLD#2_model) by using blackjax api. I've used the [SGLD](https://github.com/blackjax-devs/blackjax/blob/main/blackjax/sgmcmc/sgld.py) class. I tried to follow the example notebook that shows how to use SGMCMC in blackjax, but I can't seem to reproduce the results in the tensorflow example. 

### Steps/Code to Reproduce

I've implemented the blackjax version of the example in a Google Colab notebook located [here](https://colab.research.google.com/drive/1r7mN3ZKjDD6imlXaAUMMAHeVTG3qKR4J?usp=sharing)

### Expected Results

Samples from the posterior where there are three clusters

### Actual Results

I'm getting `nan` values for the variables I'm trying to sample.

### Versions

BlackJAX 0.8.2

### Additional comment

I suspect there are two causes of the issue:
     1. My definition of the log probability function
     2. the usage of the step size scheduling function

**Comment by rlouf** [comment]
Thank you for opening an issue. To help me debug, could you do the following:
1 - Look at the output of `logprior` and `loglikelihood` with test values for the parameters;
2 - Do the the same for the gradient estimator.

**Comment by rlouf** [comment]
The step size is probably too small. Moving this to discussions as it does not seem to be a bug. Have you tried the parameters given in the TFP article?

```python
# Learning rates and decay
starter_learning_rate = 1e-6
end_learning_rate = 1e-10
decay_steps = 1e4

# Number of training steps
training_steps = 10000

# Mini-batch size
batch_size = 20

# Sample size for parameter posteriors
sample_size = 100
```
